src/app/qgsguivectorlayertools.h

```diff
@@ -79,6 +79,26 @@ class QgsGuiVectorLayerTools : public QgsVectorLayerTools
      */
     bool saveEdits( QgsVectorLayer *layer ) const override;
 
+    /**
+     * Copy and move features with defined translation.
+     *
+     * \param layer The layer
+     * \param request The request for the features to be moved. It will be assigned to a new feature request with the newly copied features.
+     * \param dx The translation on x
+     * \param dy The translation on y
+     * \param errorMsg If given, it will contain the error message
+     * \param topologicalEditing If TRUE, the function will perform topological
+     * editing of the vertices of \a layer on \a layer and \a topologicalLayer
+     * \param topologicalLayer The layer where vertices from the moved features of \a layer will be added
+     * \param childrenInfoMsg If given, it will contain messages related to the creation of child features
+     * \returns TRUE if all features could be copied.
+     *
+     */
+    bool copyMoveFeatures( QgsVectorLayer *layer, QgsFeatureRequest &request SIP_INOUT, double dx = 0, double dy = 0, QString *errorMsg SIP_OUT = nullptr, const bool topologicalEditing = false, QgsVectorLayer *topologicalLayer = nullptr, QString *childrenInfoMsg = nullptr ) const override;
+
+  protected:
+    virtual bool avoidIntersection( QgsVectorLayer *layer, QgsFeatureRequest &request, QString *errorMsg = nullptr ) const;
```

Comment:
No need virtual here and better to make the method private as no other class need to access to it